spec.html

```diff
@@ -32156,6 +32156,33 @@ <h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-time-zone-identifiers">
+        <h1>Time Zone Identifiers</h1>
+
+        <p>
+          Time zones in ECMAScript are described using <dfn variants="time zone identifier">time zone identifiers</dfn>, which must be Strings composed entirely of characters from the Unicode Basic Latin block.
+          Time zone identifiers are case-insensitive and so are compared using ASCII case-insensitive comparisons.
+          Built-in time zones may be <dfn variants="named time zone">named time zones</dfn>, represented by the [[Identifier]] property of the records returned by AvailableNamedTimeZoneIdentifiers.
+          They may also be <dfn variants="offset time zone">offset time zones</dfn>, represented by Strings for which IsTimeZoneOffsetString returns *true*.
+        </p>
+        <p>
+          A <dfn variants="primary time zone identifiers">primary time zone identifier</dfn> represents the preferred identifier of a named time zone, while a <dfn variants="non-primary time zone identifiers">non-primary time zone identifier</dfn> is an alternate identifier for the same named time zone.
+          The set of built-in, named time zone identifiers (both primary and non-primary) that are supported by an ECMAScript implementation are called <dfn variants="available named time zone identifier">available named time zone identifiers</dfn>.
+        </p>
+        <p>
+          At a minimum, ECMAScript implementations must support a built-in time zone with the identifier *"UTC"*, which must be primary.
+          If the return value _tz_ of SystemTimeZoneIdentifier is different from *"UTC"*, and IsTimeZoneOffsetString(_tz_) is *false*, then implementations must also support _tz_ as an available named time zone identifier, and it must be primary as well.
```

Comment:
This obligates an ECMAScript implementation to do extra work when its host environment reports a time zone that would otherwise be understood as non-primary. For example, if a host reports America/Yellowknife to an implementation understanding that to be a Link for America/Edmonton, then satisfying this requirement means that the implementation must substitute in America/Edmonton. I was about to add "_or reverse the Link and remember that reversal for the lifetime of the agent even if the host environment starts reporting something different_", but then I realized that wouldn't even be possible because the Link direction might have already been observed.

It may be better to allow SystemTimeZoneIdentifier to return a non-primary time zone identifier.

Reply:
> This obligates an ECMAScript implementation to do extra work when its host environment reports a time zone that would otherwise be understood as non-primary. 

Correct, and host environments already do that work today because 402 requires it:

<img width="755" alt="image" src="https://github.com/tc39/ecma262/assets/277214/a3e77400-5941-438f-a9d7-5e1adb5e1c60">

> It may be better to allow SystemTimeZoneIdentifier to return a non-primary time zone identifier.

Changing this existing behavior (or introducing a new divergence between 262 and 402) would be out of scope for an editorial PR, so I'd prefer to leave the current text as-is unless there's some better editorial way to say it.

For a separate normative PR (or a part of proposal-canonical-tz), I'd be open to allowing engines to return non-canonical system time zone IDs. But there was opposition to making this change when we discussed ([notes](https://github.com/tc39/ecma402/blob/master/meetings/notes-2023-05-04.md#intlsupportedvaluesoftimezone-for-utc-etcgmt-and-other-non-contintentnon-ocean-time-zone-identifiers-778:~:text=what%20should%20DefaultTimeZone%20return%20when%20OS%20time%20zone%20ID%20differs%20from%20ECMAScript%20canonical%20ID)) in TG2 recently, so it's not in proposal-canonical-tz. More context: https://github.com/tc39/proposal-canonical-tz/issues/10#issuecomment-1484280615

Summary of TG2 concerns:
* It's more important for the same version of an ECMAScript implementation to behave identically across different client machines than to have different versions of ECMAScript implementations behave the same on the same client.
* It's weird if `TimeZone.p.timeZoneId` were missing from `Intl.supportedValuesOf('timeZone')`, whose output is necessarily canonicalized & deduped because its primary use case is a UI picker of time zones where `value`s would be IDs and user-visible labels would be localized names. It would be bad if, for example, "India Standard Time" showed up in that list twice because "Asia/Calcutta" and "Asia/Kolkata" both exist. If we already have to pick one ID from the Zone and Link IDs for a particular Zone, then picking the primary ID seems the best way to pick.

Not everyone agreed that those concerns were higher priority than the advantages of supporting non-primary system IDs, but there was no consensus to change the status quo.

Personally, I'm agnostic about both of those concerns above. I'm not certain that the first one matters very much, given that time zone behavior already varies across clients. And for the second, I think `===` is already an unreliable way to compare time zone names, so (assuming proposal-canonical-tz moves forward), programmers who are writing code to determine which time zone in a UI list should be selected should be using `TimeZone.p.equals()` not `===` anyways.

But I also think it doesn't matter very much, and in a proposal-canonical-tz world it matters even less.

If you feel strongly about changing this behavior, a good place to engage would be https://github.com/tc39/proposal-canonical-tz/issues/10.

Reply:
I spoke with @gibson042 offline and we'll resolve this thread as out-of-scope to this editorial PR, and will continue the discussion in proposal-canonical-tz and with TG2.